test/UnitTest/NGSI-LD/test_casesNGSI-LD.py

```diff
@@ -0,0 +1,914 @@
+        assert r.status_code == 207
+
+#testCase 19
+'''
+  Testing to delete NGSI-LD context entity
+'''
+def test_19():
+	#to create entity
+        url=brokerIp+"/ngsi-ld/v1/entities/"
+        headers={'Content-Type' : 'application/json','Accept':'application/ld+json','Link':'<{{link}}>; rel="https://uri.etsi.org/ngsi-ld/v1/ngsi-ld-core-context.jsonld"; type="application/ld+json"'}
+        r=requests.post(url,data=json.dumps(ld_data.subdata32),headers=headers)
+        print(r.content)
+        print(r.status_code)
+
+   	#to delete
+ 	url=brokerIp+"/ngsi-ld/v1/entities/urn:ngsi-ld:Vehicle:A999"
+        headers={'Content-Type':'application/json','Accept':'application/ld+json'}
+	r=requests.delete(url,headers=headers)
+        print(r.status_code)
+        assert r.status_code == 204
+
+#testCase 20
+'''
+  Testing to delete an attribute of an NGSI-LD context entity
+'''
+def test_case20():
+        url=brokerIp+"/ngsi-ld/v1/entities/"
+        headers={'Content-Type' : 'application/json','Accept':'application/ld+json','Link':'<{{link}}>; rel="https://uri.etsi.org/ngsi-ld/v1/ngsi-ld-core-context.jsonld"; type="application/ld+json"'}
+        r=requests.post(url,data=json.dumps(ld_data.subdata11),headers=headers)
+        print(r.content)
+        print(r.status_code)
+   
+
+        url=brokerIp+"/ngsi-ld/v1/entities/urn:ngsi-ld:Vehicle:A500/attrs"
+        headers={'Content-Type' : 'application/json'}
+        r=requests.post(url,data=json.dumps(ld_data.subdata4),headers=headers)
+        print(r.content)
+        print(r.status_code)
+	
+
+	url=brokerIp+"/ngsi-ld/v1/entities/urn:ngsi-ld:Vehicle:A500/attrs/brandName1"
+        r=requests.delete(url)
+	print(r.status_code)
+        assert r.status_code == 204
+
+#testCase 20
+'''
+  Testing to delete an attribute of an NGSI-LD context entity which does not have any attribute
+'''
+def test_case20():
+        url=brokerIp+"/ngsi-ld/v1/entities/"
+        headers={'Content-Type' : 'application/json','Accept':'application/ld+json','Link':'<{{link}}>; rel="https://uri.etsi.org/ngsi-ld/v1/ngsi-ld-core-context.jsonld"; type="application/ld+json"'}
+        r=requests.post(url,data=json.dumps(ld_data.subdata18),headers=headers) 
+        print(r.content)
+        print(r.status_code)
+
+
+        url=brokerIp+"/ngsi-ld/v1/entities/urn:ngsi-ld:Vehicle:A501/attrs/brandName1"
+        r=requests.delete(url)
+        print(r.content)
+	print(r.status_code)
+        assert r.status_code == 404
+
+
+#testCase 21
+'''
+  Testing to retrieve a specific entity which is deleted 
+'''
+def test_case21():
+        url=brokerIp+"/ngsi-ld/v1/entities/urn:ngsi-ld:Vehicle:A999"
+        headers={'Content-Type' : 'application/ld+json','Accept':'application/ld+json'}
+        r=requests.get(url,headers=headers)
+        print(r.content)
+	print(r.status_code)
+        assert r.status_code == 404
+
+#testCase 22
+'''
+  Testing to retrieve a specific entity which us existing
+'''
+def test_case22():
```

Comment:
[Suggestion] Get data can also be validated. like Get is done on the basis of xyz entity ID. then in response data xyz entity id should be present. this case can also be asserted for better validation.

Reply:
Overall the changes look good. I will just merge them for now. We might need to consider how to break broker.go into different modules, because it is right now too big. Let's discuss about this in the future.